tutorials/cp-cf-understand-application-lifecycle/cp-cf-understand-application-lifecycle.md

```diff
@@ -0,0 +1,97 @@
+
+  1. _Upload_ - A record is created in Cloud Foundry for the application, source files are uploaded, and an app package is created
+
+  2. _Staging_ - The required [buildpack](https://docs.cloudfoundry.org/buildpacks/) is downloaded and a droplet is created (using the generated buildpack and app package)
+
+  3. _Startup_ - The app is scheduled to run and is started up (using the droplet) within a container
+
+You can see the full application lifecycle documented at [How Apps are Staged](https://docs.cloudfoundry.org/concepts/how-applications-are-staged.html). Using the `manifest.yml` file, you can specify what actually happens during each phase, such as what buildpacks to use, the command to be run on startup, and more. A list of all the possible manifest settings can be found in the [official documentation](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html).
+
+To see each phase in action, you can push the [cf-nodejs](https://github.com/SAP-samples/cf-sample-app-nodejs) app you worked with before. Make sure to change [random-route](https://github.com/SAP-samples/cf-sample-app-nodejs/blob/8c5da6ab1b56323a9febbc67216a12c58f1c784f/manifest.yml#L6) to `true` in the `manifest.yml` or have a unique name for the app by changing the [app name](https://github.com/SAP-samples/cf-sample-app-nodejs/blob/8c5da6ab1b56323a9febbc67216a12c58f1c784f/manifest.yml#L3) in the `manifest.yml`, and then run:
+
+```
+cd <path-to>/cf-sample-app-nodejs
+cf push
+```
+
+Once you run the command you will see lots of output print to the screen, and if you look through it carefully you can see where each phase occurs.
+
+The uploading phase begins as follows:
+
+!![Uploading](uploading-phase.png)
+
+After that, the staging phase begins. You can see the buildpacks being downloaded (since a specific buildpack is not specified it downloads all of them):
+
+!![Staging Begins](staging-phase.png)
+
+You can also see when the droplet get uploaded:
+
+!![Droplet uploaded](droplet-uploaded.png)
+
+And finally you see the starting phase:
+
+!![Application starting](starting-phase.png)
+
+> If you run `cf push` for an existing application, _all instances of that application are stopped_ and new instances are started with the code you just pushed. Users that try to access the app during this time may get various HTTP error messages, such as "404" ("file not found") and others, while the application is redeployed. The Cloud Foundry API `v3` support the so called [Rolling App Deployments](https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html) with `cf push <app-name> --strategy rolling` which allows to push an app without downtime. The API `v3` are also supported on SAP BTP.
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 2: ](The Three R's - Restarting, Re-staging, and Re-pushing)]
```

Comment:
Again, please use an action verb, "Re-start, Re-stage, and Re-push your application" or whatever is appropriate.

Reply:
This feedback has been applied.